modules/storages/app/services/storages/nextcloud_managed_folder_permissions_service.rb

```diff
@@ -0,0 +1,189 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Storages
+  class NextcloudManagedFolderPermissionsService < BaseService
+    using Peripherals::ServiceResultRefinements
+
+    FILE_PERMISSIONS = OpenProject::Storages::Engine.external_file_permissions
+
+    def self.i18n_key = "nextcloud_sync_service"
+
+    class << self
+      def call(storage:, project_storages_scope: nil)
+        new(storage:, project_storages_scope:).call
+      end
+    end
+
+    def initialize(storage:, project_storages_scope: nil)
+      super()
+      @storage = storage
+      @project_storages = project_storages_scope || storage.project_storages
+    end
+
+    def call
+      with_tagged_logger([self.class.name, "storage-#{@storage.id}"]) do
+        apply_permissions_to_folders
+        epilogue
+      end
+    end
+
+    private
+
+    def epilogue
+      @result
+    end
+
+    def apply_permissions_to_folders
+      info "Setting permissions to project folders"
+      remote_admins = admin_remote_identities.pluck(:origin_user_id)
+
+      @project_storages
+        .active
+        .automatic
+        .where.not(project_folder_id: nil)
```

Comment:
WDYT about making this into a `ProjectStorage` scope?

Reply:
I am starting to wonder whether these checks are working as intended. I remember from a discussion on Matrix a while ago that missing project folders are indicated by an _empty_ string, not by nil.

I can also confirm that experimentally in my local database. So should I change the condition to exclude `""`as well?

Reply:
P.S.: See my latest commit on how I approached the new scope